pos_product_template/static/src/js/ppt.js

```diff
@@ -0,0 +1,565 @@
+            }
+            this.filter_variant();
+        },
+
+        filter_variant: function(){
+            value_list = []
+            for (var item in this.filters){
+                value_list.push(parseInt(this.filters[item]));
+            }
+            this.filter_variant_list = [];
+            for (index in this.variant_list){
+                variant = this.variant_list[index];
+                found = true;
+                for (var i = 0; i < value_list.length; i++){
+                    found = found && (variant.attribute_value_ids.indexOf(value_list[i]) != -1);
+                }
+                if (found){
+                    this.filter_variant_list.push(variant);
+                }
+            }
+            this.renderElement();
+        },
+
+        set_variant_list: function(variant_list){
+            this.variant_list = variant_list;
+            this.filter_variant_list = variant_list;
+            this.renderElement();
+        },
+
+        render_variant: function(variant){
+            var variant_html = QWeb.render('VariantWidget', {
+                    widget:  this,
+                    variant: variant,
+                });
+            var variant_node = document.createElement('div');
+            variant_node.innerHTML = variant_html;
+            variant_node = variant_node.childNodes[1];
+            return variant_node;
+        },
+
+        renderElement: function() {
+            var self = this;
+            var el_html  = openerp.qweb.render(this.template, {widget: this});
+            var el_node = document.createElement('div');
+            el_node.innerHTML = el_html;
+            el_node = el_node.childNodes[1];
+            if(this.el && this.el.parentNode){
+                this.el.parentNode.replaceChild(el_node,this.el);
+            }
+            this.el = el_node;
+            var list_container = el_node.querySelector('.variant-list');
+            for(var i = 0, len = this.filter_variant_list.length; i < len; i++){
+                var variant_node = this.render_variant(this.filter_variant_list[i]);
+                variant_node.addEventListener('click',this.click_variant_handler);
+                list_container.appendChild(variant_node);
+            }
+        },
+
+    });
+
+/* ********************************************************
+Define: pos_product_template.AttributeListWidget
+
+    - This widget will display a list of Attribute;
+*********************************************************** */
+    module.AttributeListWidget = module.PosBaseWidget.extend({
+        template:'AttributeListWidget',
+
+        init: function(parent, options) {
+            var self = this;
+            this.attribute_list = [];
+            this.product_template = null;
+            this.click_set_attribute_handler = function(event){
+                /*TODO: Refactor this function with elegant DOM manipulation */
+                // remove selected item
```

Comment:
Yes, I tried, but I didn't manage with JQuery. But I'm not JQuery Expert. Maybe could we accept this PR like that, as it's not a bug but only a 'not very clear code'. What do you think about that point ?